src/controllers/controller.bar.js

```diff
@@ -294,8 +304,39 @@ module.exports = function(Chart) {
 			model.backgroundColor = custom.backgroundColor ? custom.backgroundColor : helpers.getValueAtIndexOrDefault(dataset.backgroundColor, index, rectangleElementOptions.backgroundColor);
 			model.borderColor = custom.borderColor ? custom.borderColor : helpers.getValueAtIndexOrDefault(dataset.borderColor, index, rectangleElementOptions.borderColor);
 			model.borderWidth = custom.borderWidth ? custom.borderWidth : helpers.getValueAtIndexOrDefault(dataset.borderWidth, index, rectangleElementOptions.borderWidth);
-		}
+		},
+
+		/**
+		 * @deprecated
+		 * @private
+		 */
+		calculateBarWidth: function(ruler) {
```

Comment:
@etimberg I'm not sure I would keep these internal deprecated `calculateBar*` methods: they are not called anymore by the new implementation and most of the time people might have overridden them (which will obviously not work). The new implementation is more optimized and trying to keep compatibility will be slower (and dirty), so I don't think it worth the hack for (a few?) custom implementations using private code (e.g #2726, #3858).

What do you think?

/cc @daviddevore11 @yoerivdm